### PR TITLE
do not require AIPY for most functions

### LINKED ISSUE #527: Bury aipy imports

### PR DESCRIPTION
**Pull request by steven-murray**
Closes #527.

Made sure to change the `setup.py` and `README` as well. 

We should consider adding a set of tests that test user-based installs (without necessarily running the test-suite). I think this could be achieved reasonably easily with `tox`.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,13 +47,18 @@ def package_files(package_dir, subdirectory):
...
         'uvtools @ git+git://github.com/HERA-Team/uvtools',
         'linsolve @ git+git://github.com/HERA-Team/linsolve',
         'hera_qm @ git+git://github.com/HERA-Team/hera_qm',
         'scikit-learn'
     ],
+    'extras_require': {
+        "all": [
+            'aipy>=3.0rc2',
+        ]
+    }
```
Comment (plaplant) [inline setup.py]: I think we need a trailing comma on this line to fix the installation issues.

Comment (steven-murray) [reply setup.py]: yeah... :man_facepalming:

**Comment by plaplant** [comment]
Also, if we wanted to set up the `tox` tests you mentioned as part of this, I'm all for that.

**Comment by steven-murray** [comment]
I've tried to fix the uvtools imports as well, but I'm not sure doing `tox` in this PR is best. I think it might take a little thinking, and I know people are wanting this PR to be merged. I'll create an issue for it.

**Comment by plaplant** [comment]
Hmm, looks like tests are failing because importing `hera_sim` is broken... did something change recently on hera_sim?

**Comment by steven-murray** [comment]
Yup, and it's hopefully now fixed. I've restarted the build.

**Comment by steven-murray** [comment]
@plaplant do you need to approve? I can't tell.

**Pull request merged**